lib/mpool/src/mpool.c

```diff
@@ -83,6 +86,29 @@ mclass_params_init(
     return 0;
 }
 
+static merr_t
+mclass_path_check(enum mpool_mclass mclass, const char *path)
+{
+    merr_t err;
+    bool isdax;
+
+    if (path[0] == '\0')
+        return 0;
+
+    err = dax_path_is_fsdax(path, &isdax);
+    if (err)
+        return err;
+
+    if ((!isdax && mclass == MP_MED_PMEM) || (isdax && mclass < MP_MED_PMEM)) {
+        log_err("%s mclass path %s %s reside on a DAX filesystem",
+                mpool_mclass_to_string[mclass], path,
+                (mclass == MP_MED_PMEM) ? "must" : "must not");
+        return merr(ENOTSUP);
+    }
```

Comment:
Cleaner IMO is to have the conditional read "isdax if and only if mclass==MP_MED_PMEM":
```
if (isdax != (mclass == MP_MED_PMEM)) {
    handle error;
}
```

Reply:
I love it, will change, thnx!